pyuvsim/uvsim.py

```diff
@@ -427,6 +427,12 @@ def run_uvdata_uvsim(input_uv, beam_list, beam_dict=None, catalog=None, quiet=Fa
     # Construct beam objects from strings
     beam_list.set_obj_mode(use_shared_mem=True)
 
+    # In case the user created the beam list without checking consistency:
+    beam_list.check_consistency()
+
+    if beam_list.beam_type != 'efield':
+        raise ValueError("Beam type must be efield!")
```

Comment:
I think we should cover this line. I don't see anything like it in the main branch, so I think it was added in this PR.